### PR TITLE
operator stub, built with kubebuilder v2

### PR DESCRIPTION
**Pull request by frapposelli**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: PROJECT [hunk 1]
```diff
@@ -0,0 +1,7 @@
+version: "2"
+domain: sigs.k8s.io
```
Comment (vincepri) [inline PROJECT]: `cluster.x-k8s.io`?

**Review by vincepri — COMMENTED** [review]
Question about the apigroup/domain

File: PROJECT (hunk 1 above)
Comment (frapposelli) [reply PROJECT]: is `cluster.x-k8s.io` the same used by cluster api?

Comment (frapposelli) [reply PROJECT]: btw, this was just used as a placeholder, but happy to have a sane default from the get go

Comment (vincepri) [reply PROJECT]: That's the one we're using today in CAPI. `x-k8s.io` is the official domain for k8s experimental projects 😄

File: config/manager/manager.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+kind: Namespace
+metadata:
+  labels:
+    control-plane: controller-manager
+  name: system
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: controller-manager
```
Comment (timothysc) [inline config/manager/manager.yaml]: we should update this so it doesn't overlap

File: PROJECT (hunk 1 above)
Comment (timothysc) [reply PROJECT]: same comment.

File: api/v1/provider_types_test.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+limitations under the License.
+*/
+
+package v1
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	"golang.org/x/net/context"
```
Comment (timothysc) [inline api/v1/provider_types_test.go:22]: Is this boilerplate auto-generated?

**Review by timothysc — CHANGES_REQUESTED** [review]

File: PROJECT (hunk 1 above)
Comment (frapposelli) [reply PROJECT]: Cool, will amend

File: api/v1/provider_types_test.go (hunk 3 above)
Comment (frapposelli) [reply api/v1/provider_types_test.go:22]: Yep, auto generated by kubebuilder

File: config/manager/manager.yaml (hunk 2 above)
Comment (frapposelli) [reply config/manager/manager.yaml]: Good catch, will do

**Comment by frapposelli** [comment]
comments addressed @timothysc @vincepri

**Comment by frapposelli** [comment]
this should be good to go @timothysc

**Review by timothysc — APPROVED** [review]
/lgtm 
/approve

**Pull request merged**